lib/core/crypto.js

```diff
@@ -123,13 +132,27 @@ module.exports = (config) => {
         authTagLength: config.authTagLength,
       }
     );
-
-    if (curve === PRIME256V1) {
+    if (curve === PRIME256V1 || (curve === SECP256K1 && hasMetadata)) {
       cipher.setAAD(Buffer.from(ecdhTeamKey, 'base64'));
     }
 
+    let result;
+    if (hasMetadata) {
+      let encryptionMetadata = {
+        dr: metadata.role, // Data role
+        eo: 5, // Encryption origin
+        et: Math.round(Date.now() / 1000), // Encryption timestamp
+      };
+      const metadataBytes = pack(encryptionMetadata);
+      let offsetBuffer = Buffer.allocUnsafe(2);
+      offsetBuffer.writeUInt16LE(metadataBytes.length);
+      result = Buffer.concat([offsetBuffer, metadataBytes, Buffer.from(str)]);
+    } else {
+      result = Buffer.from(str);
+    }
```

Comment:
Extract to a method, return a Buffer, write a test

Reply:
![This is the way](https://media2.giphy.com/media/stnjSj2vpLcM4rwmEH/giphy.gif?cid=6104955epjyxn1dg96dgujogx38szm75bwmrp46p0rbumc11&ep=v1_gifs_translate&rid=giphy.gif&ct=g)